db/view/view.cc

```diff
@@ -259,15 +261,34 @@ void stats::register_stats() {
 
 bool partition_key_matches(const schema& base, const view_info& view, const dht::decorated_key& key, gc_clock::time_point now) {
```

Comment:
The `now` parameter seems redundant now.  Maybe it can be reaped in another commit?

Reply:
Hm, now I wonder if it's correct that we just ignore `now`... I'd need to think if it's possible to get a dead row here (e.g. one which has a dead row marker). If it is, then this patch is incorrect, and if it isn't, then the `now` parameter is redundant indeed. I'll leave this verification for tomorrow.

Reply:
It looks ok - or at least consistent with previous code. Previously we indeed checked if cells are alive, but only for regular cells. Keys were assumed to be there, which also means that we assume the whole row is alive or otherwise it wouldn't even reach this layer. It also kind of makes sense, because why would anyone try to apply a filter on a dead row ¯\_(ツ)_/¯

Reply:
`now` is used to convert atomic_cell to values, so if we do the conversion earlier, we don't need it here.